### PR TITLE
Remove legacy navigation (Fixes #6377)

### LINKED ISSUE #6377: Remove legacy navigation from bedrock
Once we're happy that the new Protocol navigation is here to stay, we need to remove both the switch and the code for the legacy nagivation from bedrock.

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
- Removes legacy global navigation shown to English locales.

## Issue / Bugzilla link
#6377

## Testing
Demo: https://bedrock-demo-navigation.oregon-b.moz.works/en-US/

Sucessful test run: https://ci.us-west.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/216/pipeline/

### CODE REVIEW

**Comment by alexgibson** [comment]
Ok, I think this is ready. This cleans up a bunch of leftovers and also reduces the number of requests for every page, so would be great to get in.

File: media/js/base/protocol/init-navigation.js [hunk 1]
```diff
@@ -3,26 +3,22 @@
...
     if (typeof Mzp === 'undefined' || typeof Mzp.Menu === 'undefined' || typeof Mzp.Navigation === 'undefined') {
         return;
     }
 
-    if (!Mzp.Menu.isSupported()) {
-        return;
-    }
-
-    var _mqWideNav = window.matchMedia('(min-width: 768px)');
+    var hasMediaQueries = typeof window.matchMedia !== 'undefined';
```
Comment (stephaniehobson) [inline media/js/base/protocol/init-navigation.js]: I was more comfortable with `Mzp.Menu.isSupported`. That way if the menu's requirements change the check in init is still valid.

File: bedrock/base/templates/includes/global-nav.html [hunk 2]
```diff
@@ -1,295 +0,0 @@
...
-          <title>{{_('Menu')}}</title>
-            <rect class="p1" fill="#231f20" x="3" width="5" height="6"></rect>
-            <rect class="p2" fill="#231f20" x="3" y="9.5" width="5" height="6"></rect>
-            <rect class="p3" fill="#231f20" x="3" y="19" width="5" height="6"></rect>
-        </svg>
-      </button>
-
-      <h2 class="nav-logo">
-        <a href="{{ url('mozorg.home') }}">
-          {# Bug 1438302 Avoid duplicate content for en-CA and en-GB pages. #}
```
Comment (stephaniehobson) [inline bedrock/base/templates/includes/global-nav.html]: I don't see this in the new nav file. Not a blocker for removing this code but if @rraue thinks it is important from an SEO perspective we should migrate it

File: media/js/base/protocol/init-navigation.js (hunk 1 above)
Comment (alexgibson) [reply media/js/base/protocol/init-navigation.js]: I realized that `Menu` calls `isSupported()` during `init()` anyway, so I think this was somewhat redundant.

File: bedrock/base/templates/includes/global-nav.html (hunk 2 above)
Comment (alexgibson) [reply bedrock/base/templates/includes/global-nav.html]: Oh good call, I'll update to make sure we don't lose this 👍

**Comment by alexgibson** [comment]
> Not going to hit the merge button yet in case one of the others tagged for review wants to take a peek. Let's merge and push Wednesday morning if no one else has commented by then.

@stephaniehobson If possible, it would be great to try and push this sooner rather than later. I don't forsee any issues and this should be pretty safe, but getting this pushed out today would give us a whole week to let things settle before all-hands?

File: bedrock/base/templates/includes/global-nav.html (hunk 2 above)
Comment (alexgibson) [reply bedrock/base/templates/includes/global-nav.html]: Updated to make sure we preserve this in the new navigation.

**Comment by stephaniehobson** [comment]
Needs a rebase, then sure, we can get it out today.

**Comment by alexgibson** [comment]
@stephaniehobson rebased, thank you!

**Pull request merged**